core/config/general_config.go

```diff
@@ -484,6 +486,32 @@ func (c *generalConfig) AppID() uuid.UUID {
 	return c.appID
 }
 
+// Create the audit logger configuration to send events to an
+// external service
+func (c *generalConfig) AuditLoggerConfig() *audit.AuditLoggerConfig {
+	forwardToUrl := c.viper.GetString(envvar.Name("AuditLoggerForwardToUrl"))
+	// We have this check here to determine if we should enable the audit logger
+	// at all. If this is not set, then we don't need to output the error below
+	// when configuration fails.
+	if forwardToUrl == "" {
+		return nil
+	}
+
+	auditLoggerConfig, err := audit.NewAuditLoggerConfig(
+		forwardToUrl,
+		c.viper.GetBool(envvar.Name("Dev")),
+		c.viper.GetString(envvar.Name("AuditLoggerJsonWrapperKey")),
+		c.viper.GetString(envvar.Name("AuditLoggerHeaders")),
+	)
+
+	if err != nil {
+		c.lggr.Errorf("Audit logger configuration failed with: %s", err)
+		return nil
```

Comment:
Can this cause panics elsewhere if they are expecting a non-nil configuration?

Reply:
This is only used when building the AuditLogger which also take a pointer and checks if it's nil

Reply:
Quietly disabling a system that is accidentally misconfigured is an anti-pattern that we have been trying to avoid.

A lot of this could be avoided if the individual config options had their own methods like other systems (with errors when appropriate). Then the constructor could handle things directly instead of having this delayed/deffered indirection.

Reply:
Or instead of that refactor ^ why not just return an error here also? That seems like most of the same benefits without the mess.

Reply:
How do we want to handle the difference between:
* No configuration - A valid state
* Bad configuration - An invalid state

In Rust parlance I would use the type `Result<Option<AuditLoggerConfig>, ConfigurationError>>` which is what I tried to do here with the pointer forming the option

Reply:
It would be OK to use nil config for 'disabled':
```go
func foo() (*Config, error) {
  // no config
  return nil, nil

  // bad config
  return nil, errors.Wrapf(...

  // happy path
  return &Config{...}, nil
}
```
But we also commonly have a very explicit interface method for `XXXEnabled() bool`, even if the implementation is derived from the presence of some other var(s), so this signature could be simpler:
```go
func foo() (Config, error) {
  // bad config
  return Config{}, errors.Wrapf(...

  // happy path
  return Config{...}, nil
}
```
Then the caller could check for `Config.Enabled()`, or infer it themselves.

Reply:
No longer using pointers